### ISSUE TITLE
nxos_hsrp.py (ansible 2.3.0.0) Invalid command message although configuration success

### ISSUE THREAD

**Issue by tineg** [ansible/ansible#25946]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
ansible/lib/ansible/modules/network/nxos/nxos_hsrp.py

##### ANSIBLE VERSION
ansible 2.3.0.0
  config file = /home/foo/ansible/ansible.cfg
  configured module search path = [u'/home/foo/ansible/my_modules/']
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]

##### CONFIGURATION
retry_files_enabled = False

##### OS / ENVIRONMENT
Linux myhost 3.16.0-4-amd64 #1 SMP Debian 3.16.43-2 (2017-04-30) x86_64 GNU/Linux
 NXOS: version 7.0(3)I6(1)
 NXOS: version 7.0(3)I5(2)

##### SUMMARY
hsrp parameters are properly configured but a failure warning raise and the play recap displays "changed=0" despite of the configurations entered.

##### STEPS TO REPRODUCE

---
```
foo@myhost:~$ cat test_hsrp.yml

- hosts: n9k3-1
  connection: local
  gather_facts: no

  tasks:
    - name: command testing
      nxos_hsrp:
        group: 3000
        vip: 20.20.20.254
        priority: 150
        interface: vlan3000
        preempt: enabled
        state: present
        host: "{{ ansible_host }}"
        transport: cli
        username: "{{ ansible_user }}"
        ssh_keyfile: "{{ ansible_ssh_private_key_file }}"
      register: result


    - name: debug
      debug:
        msg: "{{ result }}"
```

##### EXPECTED RESULTS
```
foo@myhost:~$ ansible-playbook test_hsrp.yml

PLAY [n9k3-1] *******************************************************************************************************************************************************

TASK [command testing] ***************************************************************************************************************************************************************
changed: [n9k3-1] 


TASK [debug] *************************************************************************************************************************************************************************
task path: /home/foo/test_hsrp.yml:22
ok: [n9k3-1]  => {
    "changed": true,
    "msg": {
        "changed": true,
        "commands": [
            "interface vlan3000",
            "hsrp version 2",
            "hsrp 3000",
            "priority 150",
            "ip 20.20.20.254",
            "preempt"
        ]
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP ***************************************************************************************************************************************************************************
n9k3-1        : ok=2    changed=1    unreachable=0    failed=0

```

##### ACTUAL RESULTS (First Play)
```


foo@myhost:~$ ansible-playbook test_hsrp.yml

PLAY [n9k3-1] *******************************************************************************************************************************************************

TASK [command testing] ***************************************************************************************************************************************************************
fatal: [n9k3-1] : FAILED! => {"changed": false, "failed": true, "msg": "interface vlan3000\r\r\n                           ^\r\n% Invalid command at '^' marker.\r\n\rn9k3-1# "}

PLAY RECAP ***************************************************************************************************************************************************************************
n9k3-1                           : ok=0    changed=0    unreachable=0    failed=1

```
##### ACTUAL RESULTS (2nd Play)
```
foo@myhost:~$ ansible-playbook test_hsrp.yml

PLAY [n9k3-1] *******************************************************************************************************************************************************

TASK [command testing] ***************************************************************************************************************************************************************
 [WARNING]: argument username has been deprecated and will be removed in a future version

 [WARNING]: argument host has been deprecated and will be removed in a future version

 [WARNING]: argument ssh_keyfile has been deprecated and will be removed in a future version

ok: [n9k3-1]

TASK [debug] *************************************************************************************************************************************************************************
ok: [n9k3-1] => {
    "changed": false,
    "msg": {
        "changed": false,
        "end_state": {
            "auth_string": "cisco",
            "auth_type": "text",
            "group": "3000",
            "interface": "vlan3000",
            "preempt": "enabled",
            "priority": "150",
            "version": "2",
            "vip": "20.20.20.254"
        },
        "existing": {
            "auth_string": "cisco",
            "auth_type": "text",
            "group": "3000",
            "interface": "vlan3000",
            "preempt": "enabled",
            "priority": "150",
            "version": "2",
            "vip": "20.20.20.254"
        },
        "proposed": {
            "group": "3000",
            "preempt": "enabled",
            "priority": "150",
            "version": "2",
            "vip": "20.20.20.254"
        },
        "updates": [],
        "warnings": [
            "argument username has been deprecated and will be removed in a future version",
            "argument host has been deprecated and will be removed in a future version",
            "argument ssh_keyfile has been deprecated and will be removed in a future version"
        ]
    }
}

PLAY RECAP ***************************************************************************************************************************************************************************
n9k3-1    : ok=2    changed=0    unreachable=0    failed=0


```

##### ACTUAL RESULTS on Nexus 93108T-EX
```
n9k3-1# sh run int vlan 3000
(...)
interface Vlan3000
  description TEST HSRP
  no shutdown
  ip address 20.20.20.252/24
  hsrp version 2
  hsrp 3000
    preempt
    priority 150
    ip 20.20.20.254

```

**Comment by trishnaguha** [comment]
@tineg This is supposed to be fixed on devel 2.4. Can you please give the module a try on devel branch?

**Comment by trishnaguha** [comment]
Closing the issue as this is already fixed in 2.4.
